crud-server/src/routes/api/conduits.js

```diff
@@ -76,7 +76,7 @@ router.patch('/:id', auth.required, async (req, res, next) => {
     if (!conduit) {
       return res.sendStatus(404);
     }
-    if (req.body.conduit.curi === undefined) {
+    if (!req.body.conduit.curi) {
       await conduit.update(await req.body.conduit);
       res.status(200).json({ conduit: conduit.toJSON() });
     } else {
```

Comment:
This code as written would result in the wrong status code returned....

if (!conduit || req.body.conduit.curi) {
  return res.sendStatus(404)
}

Reply:
// I stand corrected... here is a better version
if (!conduit) {
return res.sendStatus(404);
}

if (req.body.conduit.curi) {
  return res.sendStatus(400);
}

Reply:
I have changed the code block to check for *curi* presence in API, then do the update.
if (!conduit) {
      return res.sendStatus(404);
    }
    if (req.body.conduit.curi) {
      return res.sendStatus(400);
    }

    await conduit.update(await req.body.conduit);
    res.status(200).json({ conduit: conduit.toJSON() });